File: Tests/Resources/ExpectedTypeScriptCode/Enums/CollectionSortOrder.ts
export enum CollectionSortOrder {

  /**
   * Alphabetically, in ascending order (A - Z).
   */
  ALPHA_ASC = "ALPHA_ASC",

  /**
   * Alphabetically, in descending order (Z - A).
   */
  ALPHA_DESC = "ALPHA_DESC",

  /**
   * By best-selling products.
   */
  BEST_SELLING = "BEST_SELLING",

  /**
   * By date created, in ascending order (oldest - newest).
   */
  CREATED = "CREATED",

  /**
   * By date created, in descending order (newest - oldest).
   */
  CREATED_DESC = "CREATED_DESC",

  /**
   * In the order set manually by the merchant.
   */
  MANUAL = "MANUAL",

  /**
   * By price, in ascending order (lowest - highest).
   */
  PRICE_ASC = "PRICE_ASC",

  /**
   * By price, in descending order (highest - lowest).
   */
  PRICE_DESC = "PRICE_DESC",

  /**
   * Unknown Syrup enum.
   */
  UNKNOWN_SYRUP_ENUM = "UNKNOWN_SYRUP_ENUM"
}

File: Tests/Resources/ExpectedKotlinCode/GenerateMultipleFiles/Enums/CollectionSortOrder.kt

@file:Suppress("UNUSED_PARAMETER")
package com.shopify.syrup.enums

import com.shopify.foundation.syrupsupport.*
import com.google.gson.annotations.SerializedName
import javax.annotation.Generated

@Generated("com.shopify.syrup")
enum class CollectionSortOrder(val value: String) {

    /**
     * Alphabetically, in ascending order (A - Z).
     */
    @SerializedName("ALPHA_ASC")
    ALPHA_ASC("ALPHA_ASC"),

    /**
     * Alphabetically, in descending order (Z - A).
     */
    @SerializedName("ALPHA_DESC")
    ALPHA_DESC("ALPHA_DESC"),

    /**
     * By best-selling products.
     */
    @SerializedName("BEST_SELLING")
    BEST_SELLING("BEST_SELLING"),

    /**
     * By date created, in ascending order (oldest - newest).
     */
    @SerializedName("CREATED")
    CREATED("CREATED"),

    /**
     * By date created, in descending order (newest - oldest).
     */
    @SerializedName("CREATED_DESC")
    CREATED_DESC("CREATED_DESC"),

    /**
     * In the order set manually by the merchant.
     */
    @SerializedName("MANUAL")
    MANUAL("MANUAL"),

    /**
     * By price, in ascending order (lowest - highest).
     */
    @SerializedName("PRICE_ASC")
    PRICE_ASC("PRICE_ASC"),

    /**
     * By price, in descending order (highest - lowest).
     */
    @SerializedName("PRICE_DESC")
    PRICE_DESC("PRICE_DESC"),

    UNKNOWN_SYRUP_ENUM("UNKNOWN_SYRUP_ENUM");

    companion object {
        fun safeValueOf(name: String): CollectionSortOrder = CollectionSortOrder.values().firstOrNull { it.value == name } ?: CollectionSortOrder.UNKNOWN_SYRUP_ENUM
    }
}
